File: sdk/dotnet/Exceptions.cs
using System;

namespace FluxChat.Exceptions
{
    /// <summary>
    /// Exception levée lorsque l'API FluxChat retourne une erreur HTTP.
    /// </summary>
    public class FluxChatApiException : Exception
    {
        public int StatusCode { get; }
        public string? ApiMessage { get; }

        public FluxChatApiException(int statusCode, string? apiMessage = null)
            : base($"FluxChat API error {statusCode}: {apiMessage ?? "Unknown error"}")
        {
            StatusCode = statusCode;
            ApiMessage = apiMessage;
        }
    }

    /// <summary>
    /// Exception levée lors d'une erreur réseau (timeout, connexion refusée, etc.).
    /// </summary>
    public class FluxChatNetworkException : Exception
    {
        public FluxChatNetworkException(string message, Exception? innerException = null)
            : base(message, innerException)
        {
        }
    }
}

File: sdk/dart/lib/src/exceptions.dart
/// Exception levée lorsque l'API FluxChat retourne une erreur HTTP.
class FluxChatApiException implements Exception {
  final int statusCode;
  final String apiMessage;

  const FluxChatApiException(this.statusCode, [this.apiMessage = '']);

  @override
  String toString() =>
      'FluxChatApiException: HTTP $statusCode — $apiMessage';
}

/// Exception levée lors d'une erreur réseau (timeout, connexion refusée, etc.).
class FluxChatNetworkException implements Exception {
  final String message;
  final Object? cause;

  const FluxChatNetworkException(this.message, [this.cause]);

  @override
  String toString() => 'FluxChatNetworkException: $message'
      '${cause != null ? ' (caused by: $cause)' : ''}';
}
